### PR TITLE
[IGNORE] enable happo e2e on fork PRs

### PR DESCRIPTION
**Pull request by juliepagano**
Fixes https://github.com/perses/perses/issues/1108

# Notes for reviewers

I made this PR on a fork (instead of a branch) intentionally to test the fix worked properly.

## Validating the fix
You can tell it works by looking at the github checks.

### Before

Example of broken behavior in https://github.com/perses/perses/pull/1062. Note that only the storybook-based happo check is visible.

image

### After

In this PR, note that both happo checks are visible.

image

# Checklist

- [X] Pull request has a descriptive title and context useful to a reviewer.
- [X] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `IGNORE`.
- [X] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [X] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [X] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).

# Follow-up TODOs
- [x] Email Happo support to see if I'm missing a cleaner way to fix this and/or feature request for them to make `happo-e2e` auto-handle this case like `happo-ci` does.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -41,6 +41,13 @@ jobs:
           DEBUG: pw:webserver
           HAPPO_API_KEY: ${{ secrets.HAPPO_API_KEY }}
           HAPPO_API_SECRET: ${{ secrets.HAPPO_API_SECRET }}
+          # When a pull request is made from a fork, the secrets above are 
+          # not available for security reasons, so happo cannot use them. This 
+          # value enables happo to use a fallback behavior where it generates
+          # a one-off token for this use case.
+          # https://github.com/happo/happo-e2e/blob/master/src/loadHappoConfig.js
+          # https://github.com/happo/happo.io/blob/master/src/loadUserConfig.js
+          CHANGE_URL: ${{  github.event_name == 'pull_request' && github.event.pull_request.html_url }}
```
Comment (juliepagano) [inline .github/workflows/e2e.yml:50]: We should only need this for PRs. Merges to `main` should always have access to the secrets.

**Comment by juliepagano** [comment]
All good with the updated changes from the happo release. 

image

**Comment by Nexucis** [comment]
did you change something in the secret access from a PR ? I don't get how a bugfixe from happo changes everything :o

**Comment by juliepagano** [comment]
> did you change something in the secret access from a PR ? I don't get how a bugfixe from happo changes everything :o

They have a backup behavior where they call a happo api and create a one-time token when they're in a github actions PR that doesn't have an api key for exactly this fork use case. They had implementation for it in the code we used for the storybook tests, which is why that one worked previously. Their fix added similar behavior to the code used for e2e tests.

**Review by Nexucis — APPROVED** [review]
thanks for the explanations @juliepagano !

**Pull request merged**